### PR TITLE
Logup: more documentation

### PR DESCRIPTION
**Pull request by dannywillems**
Run `RUSTDOCFLAGS="-D warnings" cargo doc --all-features --no-deps`.

### CODE REVIEW

**Review by querolita — APPROVED** [review]
Looks great to me

File: msm/src/logup.rs [hunk 1]
```diff
@@ -1,11 +1,133 @@
...
+//! derivatives"](https://eprint.iacr.org/2022/1530.pdf) uses the sumcheck
+//! protocol to compute the partial sums (equations 16 and 17). However, we use
+//! the PlonK polynomial IOP and therefore, we will use the quotient polynomial,
+//! and the computation of the partial sums will be translated into a constraint
+//! in a new power of alpha.
+//!
+//! Note that the inner sum h(X) can be constrainted as followed:
+//! ```text
+//!         k                   k  /          k                 \
+//! h(X) *  ᴨ  (β + f_{i}(X)) = ∑  | m_{i} *  ᴨ  (β + f_{j}(X)) |     (5)
```
Comment (volhovm) [inline msm/src/logup.rs:95]: Here it should be `h(X) * \prod (\beta + \phi(i)(X))`, that is the product should include the `t`. Or you can say that `t` is just a zero (or k+1)th element alternatively.

**Review by volhovm — APPROVED** [review]
Equation 5 needs to be fixed, otherwise looks great!

**Comment by volhovm** [comment]
Oh I was too late :|

File: msm/src/logup.rs (hunk 1 above)
Comment (dannywillems) [reply msm/src/logup.rs:95]: You're right, I noticed it and I was sure someone would see it too :D. I will modify it in https://github.com/o1-labs/proof-systems/pull/2087.

**Pull request merged**